prescient/engine/egret/engine.py

```diff
@@ -373,11 +432,38 @@ def _get_solver_list(name):
         self._ruc_solver = options.deterministic_ruc_solver_type
         self._sced_solver = options.sced_solver_type
 
+        if self._ruc_solver in supported_persistent_solvers:
```

Comment:
If you wanted to get maximally cute, you could eliminate the redundancy here with a local function.

Reply:
Yeah, it's a bit tricky since we set `self._ruc_solver` and `self._sced_solver` in these two checks. It would be simpler without the try/except block, but that seems to be required. (Even invoking a non-existent PersistentSolver can raise an Exception.)

Reply:
Fair enough.